### ISSUE TITLE
 UI & Flow screens alternative login method: via code

### ISSUE THREAD

**Issue by alisonuxdesign** [ezl/dailyidea.com#469]
### This is the Visual Ui for the Alternative code login. To Add to the original ticket (create alternate login method: via code #467) 

-  Alternative Code login will ONLY be implemented on mobile. This should be the only instance in which a user is able to trigger login via code.   

-  The attached screen below shows that we will start implementation from screen 7. I have included the previous 6 screens so you can see where the 7th screen picks up in the login flow. We will go back later and address all the little UI changes later with the first 6 screens :) 

-  I have attached the UI Library guidelines for screens 7-15 as well as indicated px spacing on the first few screens for your convenience. 

- The last screen (15) is the email that will be triggered when login via code is triggered let me know how you would like this email. I can send via SVG or any other format. 

### This screenshot has spacing defined 

![Onboarding with code_spacingdefined](https://user-images.githubusercontent.com/63747360/86844123-ce61cd00-c064-11ea-9819-eea6f863c66a.png)

![Ui Buttons Mobile](https://user-images.githubusercontent.com/63747360/86843440-c81f2100-c063-11ea-880a-2c0a7c7eab80.png)

![design_system_text](https://user-images.githubusercontent.com/63747360/86843972-9195d600-c064-11ea-89df-b46c0d2f7901.png)

**Comment by vedmant** [comment]
@ezl  Few questions:

1. Should I work on this on top of the same PR as https://github.com/ezl/dailyidea.com/issues/467 , otherwise I need that merged before starting this one.
1. "Alternative Code login will ONLY be implemented on mobile." - should I detect if it's mobile based on user agent or screen size? Screen size is simpler ofcourse, but that can also work on desktop with small window.
1. As I understand in email it also should send code only for mobile requests? In this case should it be a separate email template for mobile login, without login link?

**Comment by ezl** [comment]
@vedmant -- responding to your specific questions

(cc'ing @alisonuxdesign for reference just so you can see my preference for implementation order, and will explain reasoning)

1. yes and no. put the "mobile ONLY" part in this PR. do not put the style/css in this PR. you can continue to use our generic classes for style on this new screen, and later we'll have a separate ticket for full overhaul of styles. 
2. good point regarding detection method -- for NOW, detect based on screen size. 

Reasons for excluding style from this PR (point 1 above): Makes more sense to incorporate that as a single PR rather than have it broken up into multiple PRs. (Easier to develop -- since one person will be doing that all at once. Easier to review PR -- since it's all in one place).

Reasons for decision to use screen size vs user agent -- screen size will capture all mobile clients. there's a corner case for small desktops, but the PENALTY for having a few super small desktop clients be displayed the code is very small. the reward is that it's much easier to develop, test and execute. Also a lot of effort to identify all user agents (easier to maintain). If we later choose to change the logic that determines which branch to show, it'll be easy to swap.

**Comment by vedmant** [comment]
@ezl The task is basically complete, I didn't make it pixel perfect as I see that login page doesn't match images also, I kept button yellow, the blue button has issues with disabled state. I deployed to https://beta.dailyidea.com/ please check if you want to make additional edits.

**Comment by ezl** [comment]
sounds good - @alisonuxdesign lets review this today in our meeting.

**Comment by alisonuxdesign** [comment]
hey @ezl and @vedmant I'm having trouble with this flow. I tried it on the live production site and few things need to be changed: 

1. I am not able to submit my code. The "Okay" button just doesn't submit anything and I'm stuck on the screen. 
2. when I go back to try to edit my code I need to select every single text input instead of it jumping backwards.

**Comment by ezl** [comment]
@vedmant tagging you. i reopened this issue since @alisonuxdesign reported that it didn't work as she expected.

if you feel this works as you thought it was supposed to, then the 2 of you need to discuss to make sure you are on the same page about the expected outcome.

**Comment by vedmant** [comment]
@ezl 
Hm, this looks like broken only on https://beta.dailyidea.com/ but words perfectly on my local. I'll try to re-deploy and investigate it.

@alisonuxdesign 
> when I go back to try to edit my code I need to select every single text input instead of it jumping backwards.

What do you mean by jumping backwards? 
Currently it allows to use delete button and it automatically goes between inputs: https://drive.google.com/file/d/1a1jfdp2LQ8IU9rxjP6P6tE3zp-aIZfSL/view?usp=sharing

**Comment by vedmant** [comment]
@ezl It works ok after I deployed it manually, not sure how it was deployed previous time.

**Comment by ezl** [comment]
@vedmant this is listed as in PR review, but I think it is merged -- can you confirm? If so, just move to merged in projects/1 please

**Comment by vedmant** [comment]
@ezl Yes, the issue didn't require code change

**Comment by ezl** [comment]
thanks. closing issue.
